### PR TITLE
DX: fix deprecation tests warnings for PHP 7.4

### PR DESCRIPTION
**Pull request by mvorisek**

### CODE REVIEW

File: tests/AutoReview/ProjectCodeTest.php [hunk 1]
```diff
@@ -530,7 +530,7 @@ public function testDataProvidersDeclaredReturnType(string $testClassName, \Refl
     {
         $methodId = $testClassName.'::'.$method->getName();
 
-        self::assertSame('iterable', $method->hasReturnType() ? $method->getReturnType()->__toString() : null, sprintf('DataProvider `%s` must provide `iterable` as return in method prototype.', $methodId));
+        self::assertSame('iterable', $method->hasReturnType() && $method->getReturnType() instanceof \ReflectionNamedType ? $method->getReturnType()->getName() : null, sprintf('DataProvider `%s` must provide `iterable` as return in method prototype.', $methodId));
```
Comment (mvorisek) [inline tests/AutoReview/ProjectCodeTest.php:533]: https://github.com/php/doc-en/issues/3110

File: tests/Tokenizer/Transformer/NameQualifiedTransformerTest.php [hunk 2]
```diff
@@ -108,50 +108,52 @@ public static function provideProcessCases(): iterable
...
-                new Token([T_STRING, 'Foo']),
-                new Token(';'),
-            ],
-            [
-                new Token([T_OPEN_TAG, "<?php\n"]),
-                new Token([T_NAME_FULLY_QUALIFIED, '\Foo']),
-                new Token(';'),
-            ],
-        ];
+        if (\defined('T_NAME_FULLY_QUALIFIED')) { // @TODO: drop condition when PHP 8.0+ is required
```
Comment (mvorisek) [inline tests/Tokenizer/Transformer/NameQualifiedTransformerTest.php:111]: https://github.com/sebastianbergmann/phpunit/issues/5659

Comment (Wirone) [reply tests/Tokenizer/Transformer/NameQualifiedTransformerTest.php:111]: Too bad we need to double-check this, as there's `@requires PHP 8.0` already 😕. Thanks for spotting this.

**Review by Wirone — APPROVED** [review]

**Comment by Wirone** [comment]
Thank you @mvorisek 🍻!

**Pull request merged**